src/dbnode/storage/bootstrap/result/types.go

```diff
@@ -123,7 +123,59 @@ type DatabaseSeriesBlocks struct {
 type ShardResults map[uint32]ShardResult
 
 // ShardTimeRanges is a map of shards to time ranges.
-type ShardTimeRanges map[uint32]xtime.Ranges
+type ShardTimeRanges interface {
+	// Get time ranges for a shard.
+	Get(shard uint32) xtime.Ranges
```

Comment:
Can we make this `Get(shard uint32) (xtime.Ranges, bool)`?

The amount of times we should have tested whether Get returns nil or not but forgot and then it panics (because someone didn't check whether it was nil before calling on it) is just too high to avoid making this change IMO.